cpp/daal/include/data_management/data/internal/numeric_table_sycl_csr.h

```diff
@@ -0,0 +1,673 @@
+        size_t nobs  = getNumberOfRows();
+
+        if (nfeat > 0)
+        {
+            NumericTableFeature & f = (*_ddict)[0];
+            if (isCpuTable())
+            {
+                char * data         = NULL;
+                size_t * colIndices = NULL;
+                size_t * rowOffsets = NULL;
+
+                _cpuTable->getArrays(&data, &colIndices, &rowOffsets);
+                archive->set(data, dataSize * f.typeSize);
+                archive->set(colIndices, dataSize);
+                archive->set(rowOffsets, nobs + 1);
+            }
+            else
+            {
+                services::SharedPtr<size_t> hostColIndices = _colIndices.toHost(data_management::readOnly, status);
+                DAAL_CHECK_STATUS_VAR(status);
+                services::SharedPtr<size_t> hostRowOffsets = _rowOffsets.toHost(data_management::readOnly, status);
+                DAAL_CHECK_STATUS_VAR(status);
+
+                switch (f.indexType)
+                {
+                case features::DAAL_FLOAT32:
+                {
+                    services::SharedPtr<float> hostData = _values.get<float>().toHost(data_management::readOnly, status);
+                    DAAL_CHECK_STATUS_VAR(status);
+                    archive->set(hostData.get(), dataSize);
+
+                    break;
+                }
+                case features::DAAL_FLOAT64:
+                {
+                    services::SharedPtr<double> hostData = _values.get<double>().toHost(data_management::readOnly, status);
+                    DAAL_CHECK_STATUS_VAR(status);
+                    archive->set(hostData.get(), dataSize);
+
+                    break;
+                }
+
+                default: status = services::Status(services::ErrorIncorrectParameter);
+                }
+
+                archive->set(hostColIndices.get(), dataSize);
+                archive->set(hostRowOffsets.get(), nobs + 1);
+            }
+        }
+        return services::Status();
+    }
+
+public:
+    virtual size_t getDataSize() DAAL_C11_OVERRIDE
+    {
+        if (isCpuTable())
+        {
+            return _cpuTable->getDataSize();
+        }
+
+        return _dataSize;
+    }
+
+protected:
+    template <typename DataType>
+    SyclCSRNumericTable(const services::internal::Buffer<DataType> & bufferData, const services::internal::Buffer<size_t> & bufferColIndices,
+                        const services::internal::Buffer<size_t> & bufferRowOffsets, size_t nColumns, size_t nRows, CSRIndexing indexing,
+                        services::Status & st)
+        : SyclNumericTable(nColumns, nRows, DictionaryIface::equal, st), _indexing(indexing)
+    {
+        _layout   = csrArray;
+        _dataSize = bufferData.size();
+        _defaultFeature.setType<DataType>();
+        st |= _ddict->setAllFeatures(_defaultFeature);
+
+        DAAL_ASSERT(bufferData.size() == bufferColIndices.size());
+        DAAL_ASSERT(bufferRowOffsets.size() == nRows + 1 || !_dataSize);
```

Comment:
Why asserts here and not runtime checks?

Reply:
Applied